### DISCUSSION TITLE
Expiring Daemon because JVM heap space is exhausted

### DISCUSSION THREAD

**Discussion by sergiupuhalschi** [codemagic-ci-cd/codemagic-docs#1550] [Q&A] [answered]
Hey, the build fails at minify R8 gradle task because of `ERROR:R8: java.lang.OutOfMemoryError: Java heap space`

Build id: 642f01cec9ef0d7e2ef31ca2
Gradle version: 7.5.1
AGP version: 7.4.2
gradlew and gradle-wrapper.properties are in the repository and up to date, as per [docs](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)

It builds successfully with `minifyEnabled false` and `shrinkResources false`. 

Can someone please advise on how to fix the error when enabling back the minifying and resources shrinking? Thanks!

**Comment by himesh-cm** [comment]
@sergiupuhalschi, the latest version of `Gradle` and `AGP` should do the job.
Let us know how it goes.

**Reply by sergiupuhalschi** [reply]
The latest stable Android Studio version (Electric Eel | 2022.1.1) is compatible with the [maximum AGP version of 7.4.2](https://developer.android.com/build/releases/gradle-plugin#android_gradle_plugin_and_android_studio_compatibility). Is there another way to fix the problem or it's a limitation and it requires to always use the latest & yet unstable AS version? Thanks for your reply.

**Reply by sergiupuhalschi** [reply]
Just tried with gradle 8.0.2 (but with AGP still 7.4.2) and the build failed. Build id: 64356a4d2e93d7f573ba54f5

**Reply by himesh-cm** [reply]
The minimum version with 8.0.2 is `AGP 8.0`. Please check [here](https://developer.android.com/build/releases/gradle-plugin#updating-gradle).
Please try and let us know how it goes.

**Reply by sergiupuhalschi** [reply]
Ok, but the latest stable version of AGP is [7.4.2.](https://developer.android.com/reference/tools/gradle-api) Is there another way to fix the issue without having to use preview version of AGP?

**Reply by himesh-cm** [reply, marked as answer]
Please try with the `mac_pro` or `Linux` instance and let us know how it goes.
Also, you're on a paid plan; you should contact us via the intercom widget from the bottom right corner.

**Reply by sergiupuhalschi** [reply]
This solved the issue, thank you.
